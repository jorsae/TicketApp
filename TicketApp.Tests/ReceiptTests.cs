using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Library.Model;
using ApprovalTests;
using ApprovalTests.Reporters;
using Library.Payment;

namespace TicketApp.Tests
{
    [UseReporter(typeof(VisualStudioReporter))]
    [TestFixture]
    class ReceiptTests
    {

        [Test]
        public void Assert_receipt_is_correct()
        {
            Organizer organizer = new Organizer();
            SocialEvent socialEvent = new SocialEvent(10, 100, organizer ,"TestCategory", "TestName");
            Ticket ticket = new Ticket(socialEvent);
            Vipps vipps = new Vipps();
            Receipt receipt = new Receipt(4, ticket, "kr", vipps.ToString());
            string result = receipt.GenerateReceipt();

            Approvals.Verify(result);
        }
    }
}
