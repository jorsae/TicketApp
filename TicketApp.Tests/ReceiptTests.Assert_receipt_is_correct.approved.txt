Ticket(s) for: TestName
Number of tickets: 4
Price per ticket: 100kr
Price total: 400kr
Paid using: Vipps